### ISSUE TITLE
Create custom post type templates

### ISSUE THREAD

**Issue by thomasplevy** [gocodebox/lifterlms#664]
create templates for custom post types during post type registration 

see notes in core handbook at https://wordpress.org/gutenberg/handbook/templates/

**Comment by AndreaBarghigiani** [comment]
Hi @thomasplevy,
I would like to work on this issue but which kind of block would you like to have in this CPT? And which kind of template we should create? 

I've already created some Gutenberg template for some CPT's of mine and I would like to help you out if I only have few more info about it.

**Comment by actual-saurabh** [comment]
@AndreaBarghigiani, as a first step, it'd make sense to simply replicate the current course/membership template.

![the official quickstart course for lifterlms](https://user-images.githubusercontent.com/1739834/46785419-57908680-cd4f-11e8-900d-5e61e020f0a1.png)

This would depend on the blocks that are developed inside LifterLMS from shortcodes and otherwise. Some of that would be a bit complicated to make decisions about (like the pricing table). So, it's better to focus on blocks before starting to work on templates, I guess.

**Comment by AndreaBarghigiani** [comment]
Alright, define the right blocks is essential to move the right first steps. Where do you think is the best place to list all of them. Should we list them in a separate commit and create single issues to work on them?

From this preview I can think of this list:
- [x] Paragraph (done by Gutenberg)
- [ ] Course Info
- [ ] Course Instructor
- [ ] Access Plans 
- [ ] Course Syllabus (I think we'll be better if we merge into it also the Course Outline and let decide the users via the Gutenberg Inspector)

Beside them I think those are other useful blocks in the General use of LifterLMS (I took inspiration also from your [shortcode page](https://lifterlms.com/docs/shortcodes/):
- [ ] Access Plan Button
- [ ] Course CTA Section (to drop course information into any post)
- [ ] Course Prerequisites
- [ ] Course Reviews
- [ ] Hide Content Block 
- [ ] Pricing table (I am experimenting with Gutenberg and this kind of block is on my list 😉)

I am sure that if we brainstorm it we can came up with even more blocks but I would like to better understand how we'll move on, is the process Main Issue > Single Issue the right one? How do you proceed generaly?

**Comment by actual-saurabh** [comment]
@AndreaBarghigiani The best place to start would be with existing shortcodes (#661). You could start with picking one shortcode, [creating a new issue (feature request)](https://github.com/gocodebox/lifterlms/issues/new?template=Feature_Request.md) for this shortcode.

Be sure to use JSX & ESNext since that's what we're going to use going forward. Once you're ready to send a PR, let us know in [slack](https://lifterlms.slack.com/app_redirect?channel=CCESQHE82). I'm not too sure but I guess we'd be working on a separate branch for Gutenberg compatibility before we merge it into core at some stage.

**Comment by AndreaBarghigiani** [comment]
@actual-saurabh thanks for the heads up I'll be sure to follow the steps and I think to start working on a Gutenberg branch in my fork as well.

**Comment by thomasplevy** [comment]
Moving this to [lifterlms-blocks](https://github.com/gocodebox/lifterlms-blocks), [issue #1](https://github.com/gocodebox/lifterlms-blocks/issues/1)
